packages/react/src/components/Authenticator/FederatedIdentities/context/HandleSignInWithRedirectContext.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { signInWithRedirect } from 'aws-amplify/auth';
+import { SignInWithRedirectAction } from '../controls';
+
+const HandleSignInWithRedirectContext =
+  React.createContext<SignInWithRedirectAction>(signInWithRedirect);
+
+export const HandleSignInWithRedirectProvider = ({
+  children,
+  customRedirect,
+}: {
+  children?: React.ReactNode;
+  customRedirect: SignInWithRedirectAction;
+}): JSX.Element => {
+  return (
+    <HandleSignInWithRedirectContext.Provider value={customRedirect}>
+      {children}
+    </HandleSignInWithRedirectContext.Provider>
+  );
+};
+
+export const useHandleSignInWithRedirectContext =
```

Comment:
I think I'm confused between this file and `UseHandleSigninWithRedirectContext.tsx`, they also both export `useHandleSignInWithRedirectContext` and `UseHandleSignInWithRedirectContext`.

Can they be combined in some way?

Reply:
No, they are two different context providers. I can change the names to be more clear.

The idea is that this useHandleSignInWithRedirectContext is getting the handleSignInWithRedirectContext, which is used by the buttons. the UseHandleSignInWithRedirectContext is providing context for the useHandleSignInWithRedirect hook when it is called by the customer. I don't think there is a good way to combine these.